test/unit/orientedimagelayer.js

```diff
@@ -2,69 +2,97 @@ import proj4 from 'proj4';
 import assert from 'assert';
 import OrientedImageLayer from 'Layer/OrientedImageLayer';
 import OrientedImageSource from 'Source/OrientedImageSource';
-import { HttpsProxyAgent } from 'https-proxy-agent';
 import Coordinates from 'Core/Geographic/Coordinates';
 import GlobeView from 'Core/Prefab/GlobeView';
+import sinon from 'sinon';
+import Fetcher from 'Provider/Fetcher';
 import Renderer from './bootstrap';
 
+import panoData from '../data/OrientedImage/panoramicsMetaDataParis.geojson';
+import camCalibration from '../data/OrientedImage/cameraCalibration.json';
+
+const baseurl = 'https://raw.githubusercontent.com/iTowns/iTowns2-sample-data/master/immersive';
+const orientationsUrl = `${baseurl}/exampleParis1/panoramicsMetaDataParis.geojson`;
+const calibrationUrl = `${baseurl}/exampleParis1/cameraCalibration.json`;
+
+const resources = {
+    [orientationsUrl]: panoData,
+    [calibrationUrl]: camCalibration,
+};
+
 describe('Oriented Image Layer', function () {
-    proj4.defs('EPSG:2154', '+proj=lcc +lat_1=49 +lat_2=44 +lat_0=46.5 +lon_0=3 +x_0=700000 +y_0=6600000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs');
-    const renderer = new Renderer();
+    let orImgLayer;
+    let context;
+    let stubFetcherJson;
+
+    before(function () {
+        stubFetcherJson = sinon.stub(Fetcher, 'json')
+            .callsFake(url => Promise.resolve(JSON.parse(resources[url])));
+        proj4.defs('EPSG:2154', '+proj=lcc +lat_1=49 +lat_2=44 +lat_0=46.5 +lon_0=3 +x_0=700000 +y_0=6600000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs');
+        const renderer = new Renderer();
+
+        const p = {
+            coord: new Coordinates('EPSG:4326', 2.33481381, 48.85060296),
+            range: 25,
+        };
+        const viewer = new GlobeView(renderer.domElement, p,
+            { renderer,
+                noControls: true,
+                handleCollision: false,
+                sseSubdivisionThreshold: 10,
+            });
 
-    const p = {
-        coord: new Coordinates('EPSG:4326', 2.33481381, 48.85060296),
-        range: 25,
-    };
-    const viewer = new GlobeView(renderer.domElement, p,
-        { renderer,
-            noControls: true,
-            handleCollision: false,
-            sseSubdivisionThreshold: 10,
+        // Prepare oriented image source
+        const orientedImageSource = new OrientedImageSource({
+            url: 'http://www.itowns-project.org/itowns-sample-data-small/images/140616/Paris-140616_0740-{cameraId}-00001_0000{panoId}.jpg',
+            // Url to a GEOJSON file describing feature points. It describes position and orientation of each panoramic.
+            orientationsUrl: 'https://raw.githubusercontent.com/iTowns/iTowns2-sample-data/master/' +
+                'immersive/exampleParis1/panoramicsMetaDataParis.geojson',
```

Comment:
You could use `baseurl` const

Reply:
In my opinon, this url is a parameter of ```new OrientedImageSource()``` thus it's a string and will not really be interpreted as an url (url: 'urlOrientedImage.jpg' will have work the same way). I don't really see the use of trying to reduce it at a base url name.